### ISSUE TITLE
Extend ONNX Frontend with `BatchNormalization` operators in versions 14 and 15

### ISSUE THREAD

**Issue by mbencer** [openvinotoolkit/openvino#18485]
## Context
Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.

OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.

This task requires extending OpenVINO ONNX Frontend with `BatchNormalization` operator in versions 14 and 15.

Necessary help will be provided by ONNX Fronted team.

## Operator specification
Operator details can be found in [BatchNormalization spec in version 15](https://github.com/onnx/onnx/blob/main/docs/Operators.md#batchnormalization) and [BatchNormalization spec in version 14](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#BatchNormalization-14).

## Todo list
1. In the current moment, we are supporting `BatchNormalization` [in versions 1,7,9](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/batch_norm.cpp) (note, that between versions 7 and 9 there are no differences). Prepare implementation of  versions 14 and 15 in equvalent way (in a separate opset, in form of a function)
2. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
3. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](/home/mbencer/openvino/src/frontends/onnx/tests/onnx_import.in.cpp)

More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

## Resources
- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

## Tips
Note, that in version 14 was introduced an attribute `training_mode` to determinne if an op is called in traning or inference mode (instead of basing on outputs number, like in version 9). We are not able now to handle (decompose to OV representation) this op in training mode.

## Example PRs
- https://github.com/openvinotoolkit/openvino/pull/15076
- https://github.com/openvinotoolkit/openvino/pull/14936
- https://github.com/openvinotoolkit/openvino/pull/12510

## Contact points
@mbencer
@p-wysocki
Don't hesitate to reach out, we're here to help!

Internal reference: [CVS-114253](https://jira.devtools.intel.com/browse/CVS-114253)

**Comment by Sar-thak-3** [comment]
Hi, @mbencer @p-wysocki I would love to work on this issue, I am new to Openvino, so might be requiring your great help, even with some rubbish queries but I assure to merge it soon.

**Comment by p-wysocki** [comment]
Hi @Sar-thak-3, I assigned the issue to you. Please let us know if you have any questions. :)

**Comment by Sar-thak-3** [comment]
Hi, @p-wysocki, first of all, thank you for assigning me!
I was building openvino using Docker and I run this command
```
docker build --no-cache -t image:tag .
```
in ".ci\openvino-onnx\" folder as Dockerfile is present there
But I am getting these errors, kindly help me in getting rid of it
```
 => ERROR [7/9] RUN cmake ..     -G Ninja     -DCMAKE_BUILD_TYPE=Release     -DEN  1.4s 
------
 > [7/9] RUN cmake ..     -G Ninja     -DCMAKE_BUILD_TYPE=Release     -DENABLE_INTEL_GNA=OFF     -DENABLE_INTEL_GPU=OFF     -DENABLE_HETERO=OFF     -DENABLE_MULTI=OFF     -DENABLE_AUTO_BATCH=OFF     -DENABLE_GAPI_PREPROCESSING=OFF     -DENABLE_CPPLINT=OFF     -DENABLE_NCC_STYLE=OFF     -DENABLE_PROFILING_ITT=OFF     -DENABLE_SAMPLES=OFF     -DENABLE_OV_PADDLE_FRONTEND=OFF     -DENABLE_OV_PYTORCH_FRONTEND=OFF     -DENABLE_OV_TF_FRONTEND=OFF     -DENABLE_OPENVINO_DEBUG=OFF     -DCMAKE_INSTALL_PREFIX=/openvino/dist:
1.349 CMake Warning:
1.349   Ignoring extra path from command line:
1.349
1.349    ".."
1.349
1.349
1.349 CMake Error: The source directory "/openvino" does not appear to contain CMakeLists.txt.
1.349 Specify --help for usage, or press the help button on the CMake GUI.
------
Dockerfile:51
--------------------
  50 |     WORKDIR /openvino/build
  51 | >>> RUN cmake .. \
  52 | >>>     -G Ninja \
  53 | >>>     -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
  54 | >>>     -DENABLE_INTEL_GNA=OFF \
  55 | >>>     -DENABLE_INTEL_GPU=OFF \
  56 | >>>     -DENABLE_HETERO=OFF \
  57 | >>>     -DENABLE_MULTI=OFF \
  58 | >>>     -DENABLE_AUTO_BATCH=OFF \
  59 | >>>     -DENABLE_GAPI_PREPROCESSING=OFF \
  60 | >>>     -DENABLE_CPPLINT=OFF \
  61 | >>>     -DENABLE_NCC_STYLE=OFF \
  62 | >>>     -DENABLE_PROFILING_ITT=OFF \
  63 | >>>     -DENABLE_SAMPLES=OFF \
  64 | >>>     -DENABLE_OV_PADDLE_FRONTEND=OFF \
  65 | >>>     -DENABLE_OV_PYTORCH_FRONTEND=OFF \
  66 | >>>     -DENABLE_OV_TF_FRONTEND=OFF \
  67 | >>>     -DENABLE_OPENVINO_DEBUG=OFF \
  68 | >>>     -DCMAKE_INSTALL_PREFIX=/openvino/dist
  69 |     RUN ninja install
--------------------
ERROR: failed to solve: process "/bin/sh -c cmake ..     -G Ninja     -DCMAKE_BUILD_TYPE=${BUILD_TYPE}     -DENABLE_INTEL_GNA=OFF     -DENABLE_INTEL_GPU=OFF     -DENABLE_HETERO=OFF     -DENABLE_MULTI=OFF     -DENABLE_AUTO_BATCH=OFF     -DENABLE_GAPI_PREPROCESSING=OFF     -DENABLE_CPPLINT=OFF     -DENABLE_NCC_STYLE=OFF     -DENABLE_PROFILING_ITT=OFF     -DENABLE_SAMPLES=OFF     -DENABLE_OV_PADDLE_FRONTEND=OFF     -DENABLE_OV_PYTORCH_FRONTEND=OFF     -DENABLE_OV_TF_FRONTEND=OFF     -DENABLE_OPENVINO_DEBUG=OFF     -DCMAKE_INSTALL_PREFIX=/openvino/dist" did not complete successfully: exit code: 1
```

**Comment by p-wysocki** [comment]
I'm returning the task to being open due to current assignee's inactivity.

**Comment by siddhant-0707** [comment]
Hey @p-wysocki may I work on this

**Comment by p-wysocki** [comment]
Hello @siddhant-0707, sure! I assigned you.

**Comment by p-wysocki** [comment]
@siddhant-0707 could you please confirm whether you're still working on this or can I mark this issue as open for pickup?

**Comment by siddhant-0707** [comment]
@p-wysocki Yes I'm almost done with this, sorry for the delay. I'll soon open a PR.

**Comment by p-wysocki** [comment]
No problem, I'm just refreshing the task statuses in the main good first issue list. :)

**Comment by p-wysocki** [comment]
@siddhant-0707 are you still working on this or can I return the issue to be picked up by other contributors?

**Comment by siddhant-0707** [comment]
Hey, I'm occupied by other issues. I'll take it up another time if available. Apologies for the inconvenience.

**Comment by YashwardhanSingh2005** [comment]
@p-wysocki hey i was working on this issue just wanted a little clarity that should i create the same set for both op 14 and 15 or diff sets also we still only support inference mode right?

**Comment by mlukasze** [comment]
@gkrivor would way do you prefer

**Comment by YashwardhanSingh2005** [comment]
hey @mbencer @p-wysocki  i wanted to test my changes and have created my test model which test exactly do i execute? the docs  have various tests Do i run the python operator test to test my operator?

**Comment by YashwardhanSingh2005** [comment]
.take

**Comment by YashwardhanSingh2005** [comment]
I have created a PR [#21850](https://github.com/openvinotoolkit/openvino/pull/21850) please review it.

**Comment by p-wysocki** [comment]
Hello @YashwardhanSingh2005, thank you for your contribution! Is there anything you need help with? I see that in the PR you mentioned some issues.

I am happy to announce that we have created a channel dedicated to Good First Issues support on our [Intel DevHub Discord server](https://discord.gg/wPuqAujS)! Join it to receive support, engage in discussions, ask questions and talk to OpenVINO developers.

**Comment by YashwardhanSingh2005** [comment]
Thanks! Altho i may not be able to acticely contribute rn would look forward to it in future!

**Comment by p-wysocki** [comment]
I'm returning this task into being open - your draft PR can serve as a starting point for next contributor. You're welcome back anytime!

**Comment by Mr-MVP** [comment]
Hey! @p-wysocki I've recently build OpenVINO from source and am going through the documentations of the ONNX operations and different version operations. Can you please assign this issue to me? I would love to work on it!

**Comment by Mr-MVP** [comment]
.take

**Comment by p-wysocki** [comment]
Hi @Mr-MVP, thanks for taking a look! Please let us know if you have any questions, here or on Discord. :)

**Comment by Mr-MVP** [comment]
Can you please link the discord server

**Comment by siddhant-0707** [comment]
[Discord Link](https://discord.gg/7pVRxUwdWG)

**Comment by p-wysocki** [comment]
Hello @Mr-MVP, are you still working on that issue?

**Comment by Mr-MVP** [comment]
Yes I am.

**Comment by p-wysocki** [comment]
Hello @Mr-MVP, are there any updates?

**Comment by Bepitic** [comment]
.take
